test/conformance/walk.go

```diff
@@ -130,37 +296,67 @@ func scandir(dir string) {
 	}
 }
 
-func scanfile(path string) {
+func scanfile(path string, src interface{}) []conformanceData {
 	v := newVisitor()
-	file, err := parser.ParseFile(v.FileSet, path, nil, 0)
+	file, err := parser.ParseFile(v.FileSet, path, src, parser.ParseComments)
 	if err != nil {
 		panic(err)
 	}
 
+	v.cMap = ast.NewCommentMap(v.FileSet, file, file.Comments)
+
 	ast.Walk(v, file)
+	return v.tests
 }
 
 func main() {
-	args := os.Args[1:]
-	if len(args) < 1 {
+	flag.Parse()
+
+	if len(flag.Args()) < 1 {
 		fmt.Fprintf(os.Stderr, "USAGE: %s <DIR or FILE> [...]\n", os.Args[0])
 		os.Exit(64)
 	}
 
-	for _, arg := range args {
+	var confFile *os.File
+	if *confDoc {
+		var err error
+		confFile, err = os.Create("conformance.md")
```

Comment:
Instead of writing to a hardcoded file, I would prefer we either send it to stdout or take a pathname as a flag.

The easiest way to make this re-usable is to always write to stdout, but operate only in one mode or the other.  E.g., add a flag like `--output=shortlist`, `--output=markdown`, and leave the default as `shortlist` for backwards compatibility.

Reply:
I can add a command line argument to specify the path. When --conformance flag is set we always generate markdown(at least that was the decision earlier). Please let me know.

Reply:
writing to stdout.